src/test/scala/it/pagopa/interop/purposeprocess/SpecHelper.scala

```diff
@@ -331,8 +325,8 @@ trait SpecHelper extends SprayJsonSupport with DefaultJsonProtocol with MockFact
     val actualEService = eService.getOrElse(SpecData.eService).copy(descriptors = Seq(descriptor))
     mockAgreementsRetrieve(purpose.eserviceId, purpose.consumerId, Seq(agreement))
     mockEServiceRetrieve(purpose.eserviceId, actualEService)
-    mockOrganizationRetrieve(actualEService.producerId)
-    mockOrganizationRetrieve(purpose.consumerId)
+    mockOrganizationRetrieve(agreement.producerId)
+    mockOrganizationRetrieve(agreement.consumerId)
```

Comment:
is there any reason for this change?

Reply:
I answer directly here, the different changes being connected. I originally made your first suggested change but the tests didn't pass. While trying to figure out the behavior I forgot about this change. The only way to pass producerId and consumerId correctly was to modify the mock. With your second suggested change now it's all resolved, thank! 😄